op-supervisor/supervisor/service.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum-optimism/optimism/op-service/cliapp"
+	"github.com/ethereum-optimism/optimism/op-service/httputil"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/oppprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+	"github.com/ethereum-optimism/optimism/op-supervisor/metrics"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/frontend"
+)
+
+type Backend interface {
+	frontend.Backend
+	io.Closer
+}
+
+// SupervisorService implements the full-environment bells and whistles around the Supervisor.
+// This includes the setup and teardown of metrics, pprof, admin RPC, regular RPC etc.
+type SupervisorService struct {
+	closing atomic.Bool
+
+	log log.Logger
+
+	metrics metrics.Metricer
+
+	backend Backend
+
+	pprofService *oppprof.Service
+	metricsSrv   *httputil.HTTPServer
+	rpcServer    *oprpc.Server
+}
+
+var _ cliapp.Lifecycle = (*SupervisorService)(nil)
+
+func SupervisorFromCLIConfig(ctx context.Context, cfg *CLIConfig, logger log.Logger) (*SupervisorService, error) {
+	su := &SupervisorService{log: logger}
+	if err := su.initFromCLIConfig(ctx, cfg); err != nil {
+		return nil, errors.Join(err, su.Stop(ctx)) // try to clean up our failed initialization attempt
+	}
+	return su, nil
+}
+
+func (su *SupervisorService) initFromCLIConfig(ctx context.Context, cfg *CLIConfig) error {
+	su.initMetrics(cfg)
+	if err := su.initPProf(cfg); err != nil {
+		return fmt.Errorf("failed to start PProf server: %w", err)
+	}
+	if err := su.initMetricsServer(cfg); err != nil {
+		return fmt.Errorf("failed to start Metrics server: %w", err)
+	}
+	su.initBackend(cfg)
+	if err := su.initRPCServer(cfg); err != nil {
+		return fmt.Errorf("failed to start RPC server: %w", err)
+	}
+	return nil
+}
+
+func (su *SupervisorService) initBackend(cfg *CLIConfig) {
+	if cfg.MockRun {
+		su.backend = backend.NewMockBackend()
+	} else {
+		su.backend = backend.NewSupervisorBackend()
+	}
+}
+
+func (su *SupervisorService) initMetrics(cfg *CLIConfig) {
+	if cfg.MetricsConfig.Enabled {
+		procName := "default"
+		su.metrics = metrics.NewMetrics(procName)
+		su.metrics.RecordInfo(cfg.Version)
+	} else {
+		su.metrics = metrics.NoopMetrics
+	}
+}
+
+func (su *SupervisorService) initPProf(cfg *CLIConfig) error {
+	su.pprofService = oppprof.New(
+		cfg.PprofConfig.ListenEnabled,
+		cfg.PprofConfig.ListenAddr,
+		cfg.PprofConfig.ListenPort,
+		cfg.PprofConfig.ProfileType,
+		cfg.PprofConfig.ProfileDir,
+		cfg.PprofConfig.ProfileFilename,
+	)
+
+	if err := su.pprofService.Start(); err != nil {
+		return fmt.Errorf("failed to start pprof service: %w", err)
+	}
+
+	return nil
+}
+
+func (su *SupervisorService) initMetricsServer(cfg *CLIConfig) error {
+	if !cfg.MetricsConfig.Enabled {
+		su.log.Info("metrics disabled")
```

Comment:
```suggestion
		su.log.Info("Metrics disabled")
```

Reply:
I'm pushing a commit to fix all of the log capitalisation. Some hills are worth dying on. 🤣